modules/navier_stokes/src/fvbcs/INSFVSymmetryVelocityBC.C

```diff
@@ -48,36 +41,48 @@ INSFVSymmetryVelocityBC::INSFVSymmetryVelocityBC(const InputParameters & params)
 #endif
 }
 
-ADReal
-INSFVSymmetryVelocityBC::computeQpResidual()
+void
+INSFVSymmetryVelocityBC::gatherRCData(const FaceInfo & fi)
 {
+  _face_info = &fi;
+  _face_type = fi.faceType(_var.name());
+
   const bool use_elem = _face_info->faceType(_var.name()) == FaceInfo::VarFaceNeighbors::ELEM;
+  const auto elem_arg =
+      use_elem ? makeElemArg(&_face_info->elem()) : makeElemArg(_face_info->neighborPtr());
+  const auto normal = use_elem ? _face_info->normal() : Point(-_face_info->normal());
   const Point & cell_centroid =
       use_elem ? _face_info->elemCentroid() : _face_info->neighborCentroid();
-  const auto & u_C = use_elem ? _u_elem : _u_neighbor;
-  const auto & v_C = use_elem ? _v_elem : _v_neighbor;
-  const auto & w_C = use_elem ? _w_elem : _w_neighbor;
-
-  // Evaluate viscosity on the face
-  const auto mu_b = use_elem ? _mu(std::make_tuple(_face_info,
-                                                   Moose::FV::LimiterType::CentralDifference,
-                                                   true,
-                                                   _face_info->elem().subdomain_id()))
-                             : _mu(std::make_tuple(_face_info,
-                                                   Moose::FV::LimiterType::CentralDifference,
-                                                   true,
-                                                   _face_info->neighborPtr()->subdomain_id()));
+  const auto u_C = _u_functor(elem_arg);
+  const auto v_C = _v_functor(elem_arg);
+  const auto w_C = _w_functor(elem_arg);
 
-  const auto d_perpendicular = std::abs((_face_info->faceCentroid() - cell_centroid) * _normal);
+  const auto d_perpendicular = std::abs((_face_info->faceCentroid() - cell_centroid) * normal);
 
   // See Moukalled 15.150. Recall that we multiply by the area in the base class, so S_b ->
-  // _normal.norm() here
+  // normal.norm() -> 1 here.
+
+  const auto face = singleSidedFaceArg();
+  const auto mu_b = _mu(face);
+  const auto eps_b = epsFunctor()(face);
 
-  ADReal v_dot_n = u_C[_qp] * _normal(0);
+  ADReal v_dot_n = u_C * normal(0);
   if (_dim > 1)
-    v_dot_n += v_C[_qp] * _normal(1);
+    v_dot_n += v_C * normal(1);
   if (_dim > 2)
-    v_dot_n += w_C[_qp] + _normal(2);
+    v_dot_n += w_C * normal(2);
+
+  const auto strong_resid = mu_b / eps_b / d_perpendicular * normal(_index) * v_dot_n;
+
+  // The strong residual for this object is a superposition of all the velocity components, however,
+  // for the on-diagonal 'a' coefficient, we only care about the coefficient multiplying the
+  // velocity component corresponding to _index, hence v_dot_n -> normal(_index) when moving from
+  // strong_resid -> a
+  const auto a = mu_b / eps_b / d_perpendicular * normal(_index) * normal(_index);
```

Comment:
Well in general we can't bring the eps out of the grad u, but in most of our use cases eps probably won't be varying in the normal direction at boundaries so I'm on board for your proposed change. What's funny is that in the most common symmetry case, an axis at r = 0, this BC doesn't even matter because we multiply by a `_coord` that is 0